mods/other/spectator/init.lua

```diff
@@ -0,0 +1,122 @@
+-- SPDX-License-Identifier: GPL-3.0-or-later
+-- Copyright (c) 2024 Ivan Shkatov (Maintainer_) ivanskatov672@gmail.com
+spectator = {}
+spectator.in_ = {}
+
+-- Override interact privilege
+minetest.register_privilege("interact", {
+    description = "Can interact with things and modify the world",
+    give_to_admin = false,
+    give_to_singleplayer = false
+})
+
+-- Override shout privilege
+minetest.register_privilege("shout", {
+    description = "Can speak in chat",
+    give_to_admin = false,
+    give_to_singleplayer = false
+})
+
+-- Enabled spectator mode
+spectator.on = function (player)
+    local name = player:get_player_name()
+    local meta = player:get_meta()
+
+    meta:set_string(name, minetest.privs_to_string(minetest.get_player_privs(name), ","))
```

Comment:
Makes sense to store and modify only necessary privileges like fly, shout, noclip, fast 

Keep in mind that while a player is in spectator mode, players privileges cannot be changed in the normal way.
For example, you had the ban privilege, enter spectator mode, admin revoke your ban priv, you left spectator mode, the ban privilege is still with you